11-BLE-peripheral/main.c

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (C) 2025 TU Dresden
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ * @brief       WSN_BLE_peripheral_01
+ * @author      Bozheng Pang <bozheng.pang@tu-dresden.de>
+ */
+
+#include "random.h"
+#include "ztimer.h"
+#include "host/util/util.h"
+#include "host/ble_gap.h"
+#include "services/gap/ble_svc_gap.h"
+
+/* Define UUIDs. */
+#define BLE_SVC_CUSTOM_UUID                         0xff00
+#define BLE_CHR_CUSTOM_NOTIFY_UUID                  0xee00
+/* Define variables that will be used later. */
+const char *device_name = "WSN_BLE_peripheral_01";
+uint8_t addr_type;
+uint8_t conn_state;
+uint16_t conn_handle;
+uint8_t notify_state;
+/* Define characteristic value handle. */
+uint16_t custom_notify_data_val_handle;
+
+void advertise(void);
+
+/* Define characteristic access callback handler. */
+int custom_notify_data_handler(uint16_t conn_handle, uint16_t attr_handle,
+    struct ble_gatt_access_ctxt *ctxt, void *arg)
+{
+    (void)conn_handle;
+    (void)attr_handle;
+    (void)arg;
+    if (ble_uuid_u16(ctxt->chr->uuid) != BLE_CHR_CUSTOM_NOTIFY_UUID) {
+        return BLE_ATT_ERR_UNLIKELY;
+    }
+    return 0;
+}
+
+/* Define our custom service and characteristic. */
+const struct ble_gatt_svc_def gatt_svr_svcs[] = {
+    {
+        /* Custom Service */
+        .type = BLE_GATT_SVC_TYPE_PRIMARY,
+        .uuid = BLE_UUID16_DECLARE(BLE_SVC_CUSTOM_UUID),
+        .characteristics = (struct ble_gatt_chr_def[]) { {
+            /* Custom Notify Characteristic */
+            .uuid = BLE_UUID16_DECLARE(BLE_CHR_CUSTOM_NOTIFY_UUID),
+            .access_cb = custom_notify_data_handler,
+            .val_handle = &custom_notify_data_val_handle,
+            .flags = BLE_GATT_CHR_F_NOTIFY,
+        },  {
+            0, /* No more characteristics in this service */
+        }, }
+    },
+    {
+        0, /* No more services */
+    },
+};
+
+int peripheral_conn_event(struct ble_gap_event *event, void *arg)
+{
+	(void)arg;
```

Comment:
```suggestion
    (void)arg;
```

(all of them should be whitespaces)